spyder/api/plugins/enum.py

```diff
@@ -1,69 +1,160 @@
-    Debugger = 'debugger'
-    Editor = 'editor'
-    Explorer = 'explorer'
-    ExternalTerminal = 'external_terminal'
-    Find = 'find_in_files'
-    Help = 'help'
-    History = 'historylog'
-    IPythonConsole = 'ipython_console'
-    Layout = 'layout'
-    MainInterpreter = 'main_interpreter'
-    MainMenu = 'mainmenu'
-    OnlineHelp = 'onlinehelp'
-    OutlineExplorer = 'outline_explorer'
-    Plots = 'plots'
-    Preferences = 'preferences'
-    Profiler = 'profiler'
-    Projects = 'project_explorer'
-    Pylint = 'pylint'
-    PythonpathManager = 'pythonpath_manager'
-    RemoteClient = 'remoteclient'
-    Run = 'run'
-    Shortcuts = 'shortcuts'
-    StatusBar = 'statusbar'
-    Switcher = 'switcher'
+
+    Appearance = "appearance"
+    """The Spyder Appearance plugin."""
+
+    Application = "application"
+    """The Spyder Application plugin."""
+
+    Completions = "completions"
+    """The Spyder Completions plugin."""
+
+    ExternalTerminal = "external_terminal"
+    """The Spyder External Terminal plugin."""
+
+    Layout = "layout"
+    """The Spyder Layout plugin."""
+
+    MainInterpreter = "main_interpreter"
+    """The Spyder Main Interpreter plugin."""
+
+    MainMenu = "mainmenu"
+    """The Spyder Main Menu plugin."""
+
+    Preferences = "preferences"
+    """The Spyder Preferences plugin."""
+
+    PythonpathManager = "pythonpath_manager"
+    """The Spyder PYTHONPATH Manager plugin."""
+
+    RemoteClient = "remoteclient"
+    """The Spyder Remote Client plugin."""
+
+    Run = "run"
+    """The Spyder Run plugin."""
+
+    Shortcuts = "shortcuts"
+    """The Spyder Shortcuts plugin."""
+
+    StatusBar = "statusbar"
+    """The Spyder Status Bar plugin."""
+
+    Switcher = "switcher"
+    """The Spyder Switcher plugin."""
+
     Toolbar = "toolbar"
-    Tours = 'tours'
-    UpdateManager = 'update_manager'
-    VariableExplorer = 'variable_explorer'
-    WorkingDirectory = 'workingdir'
+    """The Spyder Toolbar plugin."""
 
+    Tours = "tours"
+    """The Spyder Tours plugin."""
 
-class DockablePlugins:
-    Console = 'internal_console'
-    Debugger = 'debugger'
-    Editor = 'editor'
-    Explorer = 'explorer'
-    Find = 'find_in_files'
-    Help = 'help'
-    History = 'historylog'
-    IPythonConsole = 'ipython_console'
-    OnlineHelp = 'onlinehelp'
-    OutlineExplorer = 'outline_explorer'
-    Plots = 'plots'
-    Profiler = 'profiler'
-    Projects = 'project_explorer'
-    Pylint = 'pylint'
-    VariableExplorer = 'variable_explorer'
+    UpdateManager = "update_manager"
+    """The Spyder Update Manager plugin."""
+
+    WorkingDirectory = "workingdir"
+    """The Spyder Working Directory plugin."""
 
 
 class OptionalPlugins:
+    """Pseudo-enum class listing the names of Spyder optional plugins.
+
+    Values correspond to the plugin's
+    :attr:`~spyder.api.plugins.SpyderPluginV2.NAME`.
+    """
+
     EnvManager = "spyder_env_manager"
+    """The Spyder Env Manager plugin."""
```

Comment:
```suggestion
    """The Spyder Environment Manager plugin."""
```

We can be more explicit about this plugin.